src/utils/cache.ts

```diff
@@ -0,0 +1,50 @@
+export const cachedFetch = (url: string, options: number | RequestInit & {expireInSeconds: number, forceRefresh?: boolean} | undefined) => {
+  let expiry = 5 * 60 // 5 min default
+  if (typeof options === 'number') {
+    expiry = options
+    options = undefined
+  } else if (typeof options === 'object') {
+    // I hope you didn't set it to 0 seconds
+    expiry = options.expireInSeconds || expiry
+  }
+  // Use the URL as the cache key to sessionStorage
+  let cacheKey = url
+  let cached = localStorage.getItem(cacheKey)
+  let whenCached = localStorage.getItem(cacheKey + ':ts')
+  if (cached !== null && whenCached !== null && !options?.forceRefresh) {
+    // it was in sessionStorage! Yay!
+    // Even though 'whenCached' is a string, this operation
+    // works because the minus sign converts the
+    // string to an integer and it will work.
+    let age = (Date.now() - parseInt(whenCached)) / 1000
+    if (age < expiry) {
+      let response = new Response(new Blob([cached]))
+      return Promise.resolve(response)
+    } else {
+      // We need to clean up this old key
+      localStorage.removeItem(cacheKey)
+      localStorage.removeItem(cacheKey + ':ts')
+    }
+  }
+
+  return fetch(url, options).then(response => {
+    // let's only store in cache if the content-type is
+    // JSON or something non-binary
+    if (response.status === 200) {
+      let ct = response.headers.get('Content-Type')
+      if (ct && (ct.match(/application\/json/i) || ct.match(/text\//i))) {
+        // There is a .json() instead of .text() but
+        // we're going to store it in sessionStorage as
+        // string anyway.
+        // If we don't clone the response, it will be
+        // consumed by the time it's returned. This
+        // way we're being un-intrusive.
+        response.clone().text().then(content => {
+          localStorage.setItem(cacheKey, content)
+          localStorage.setItem(cacheKey+':ts', `${Date.now()}`)
+        })
+      }
+    }
+    return response
+  })
+}
```

Comment:
```suggestion
export const cachedFetch = async (
  url: string,
  options:
    | number
    | (RequestInit & { expireInSeconds: number; forceRefresh?: boolean })
    | undefined
) => {
  let expiry = 5 * 60; // 5 min default
  if (typeof options === "number") {
    expiry = options;
    options = undefined;
  } else if (typeof options === "object") {
    expiry = options.expireInSeconds ?? expiry;
  }

  let cacheKey = url;
  let cached = (await chrome.storage.local.get(cacheKey))[cacheKey];
  let whenCached = (await chrome.storage.local.get(cacheKey + ":ts"))[cacheKey + ":ts"];
  
  if (cached && whenCached !== null && !options?.forceRefresh) {
    let age = (Date.now() - parseInt(whenCached)) / 1000;
    if (age < expiry) {
      let response = new Response(new Blob([cached]));
      return Promise.resolve(response);
    } else {
      chrome.storage.local.remove(cacheKey);
      chrome.storage.local.remove(cacheKey + ":ts");
    }
  }

  return fetch(url, options).then((response) => {
    if (response.status === 200) {
      let ct = response.headers.get("Content-Type");
      if (ct && ct.match(/(application\/json|text\/.*)/i)) {
        response
          .clone()
          .text()
          .then((content) => {
            chrome.storage.local.set({ [cacheKey]: content });
            chrome.storage.local.set({ [cacheKey + ":ts"]: Date.now() });
          });
      }
    }
    return response;
  });
};
```
Consider the following changes I've made:
* Replaced `||` with `??` to avoid a falsy result when `expiry` is set to 0 seconds.
* Replaced `localStorage` with `chrome.storage.local` .
* Combined the `text/*`, `application/json` check into a single `match()` call.
*  Formatted the file with semi-colons(Linting will be added soon, so this shouldn't be a problem).